cirrus/server/cirrus/main.py

```diff
@@ -76,9 +82,11 @@ def read_root():
 
 
 @app.post("/v1/features/", status_code=status.HTTP_200_OK)
-async def compute_features():
-    # # will recieve as part of incoming request
-    targeting_context: Dict[str, Any] = {"clientId": "test", "requestContext": {}}
+async def compute_features(request_data: FeatureRequest):
+    targeting_context: Dict[str, Any] = {
+        "clientId": request_data.client_id or "",
+        "requestContext": request_data.context or {},
```

Comment:
Should we not fail some validation step and abort here if either the `client_id` or `context` are missing?

Reply:
my idea was whatever happens with a request if the client id is provided or not provided we return the default config, definitely it will be logged whenever that happen, or we can return 400 with the below changes
```python
 if not request_data.client_id:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Missing client ID")
 if not request_data.context:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Missing context")

Reply:
Yeah my point about returning the defaults was in the case where we're unable to sync with remote settings.  A malformed request should always return an error to the caller, otherwise something could break on their end and they'd have no way to know.

Reply:
Yeah I'd expect it to return a 400.  You could do it that way but at some point we'll want to have a proper structure for deserializing and validating the data.

Reply:
okay so will be using default 422 which fastapi provides to validate